src/react/deckgl.js

```diff
@@ -89,6 +93,8 @@ export default class DeckGL extends React.Component {
 
     // Note: avoid React setState due GL animation loop / setState timing issue
     this.layerManager = new LayerManager({gl});
+    this.effectManager = new EffectManager({gl, deckgl: this});
```

Comment:
This line is a pretty big deal since it breaks the idea that deck.gl is independent of react. (you are giving EffectManager access to the react component). If you can just pass in layerManager that will be solved.